File: sdkwork-backend-sdk-go/types/plus_ai_agent_vo.go
package types

// AI Agent Value Object (Represents an AI model agent with configuration and associations)
type PlusAiAgentVO struct {
	CreatedAt string `json:"createdAt"`
	UpdatedAt string `json:"updatedAt"`
	Id int `json:"id"`
	Name string `json:"name"`
	FaceImage ImageMediaResource `json:"faceImage"`
	FaceVideo VideoMediaResource `json:"faceVideo"`
	Owner string `json:"owner"`
	OwnerId int `json:"ownerId"`
	Channel string `json:"channel"`
	ChannelId string `json:"channelId"`
	Icon string `json:"icon"`
	Description string `json:"description"`
	Tags TagsContent `json:"tags"`
	Type string `json:"type"`
	BizType string `json:"bizType"`
	Status string `json:"status"`
	BaseConfig PlusAiAgentConfig `json:"baseConfig"`
	KnowledgeConfig AgentKnowledgeConfig `json:"knowledgeConfig"`
	MemoryConfig AgentMemoryConfig `json:"memoryConfig"`
	SpeechConfig SpeechConfig `json:"speechConfig"`
	ToolConfig AgentToolConfig `json:"toolConfig"`
	Members TeamAgentMembers `json:"members"`
	KnowledgeBaseId int `json:"knowledgeBaseId"`
	PromptId int `json:"promptId"`
	CategoryId int `json:"categoryId"`
	ChatOptions PlusChatOptions `json:"chatOptions"`
	Config PlusAiAgentConfig `json:"config"`
	UsageCount int `json:"usageCount"`
}

File: sdkwork-backend-sdk-python/sdkwork_backend_sdk/models/plus_ai_agent_vo.py
from dataclasses import dataclass
from typing import Optional, List, Dict, Any

@dataclass
class PlusAiAgentVO:
    """AI Agent Value Object (Represents an AI model agent with configuration and associations)"""
    created_at: str = None
    updated_at: str = None
    id: int = None
    name: str = None
    face_image: ImageMediaResource = None
    face_video: VideoMediaResource = None
    owner: str = None
    owner_id: int = None
    channel: str = None
    channel_id: str = None
    icon: str = None
    description: str = None
    tags: TagsContent = None
    type: str = None
    biz_type: str = None
    status: str = None
    base_config: PlusAiAgentConfig = None
    knowledge_config: AgentKnowledgeConfig = None
    memory_config: AgentMemoryConfig = None
    speech_config: SpeechConfig = None
    tool_config: AgentToolConfig = None
    members: TeamAgentMembers = None
    knowledge_base_id: int = None
    prompt_id: int = None
    category_id: int = None
    chat_options: PlusChatOptions = None
    config: PlusAiAgentConfig = None
    usage_count: int = None
